Makefile

```diff
@@ -196,9 +196,9 @@ ifneq ($(SKIP_BOOTSTRAP),1)
 # If we're in a git worktree, the git hooks directory may not be in our root,
 # so we ask git for the location.
 #
-# Note that this command requires git 2.5+; earlier versions produce opaque
-# errors.
-GITHOOKSDIR := $(shell git rev-parse --git-path hooks)
+# Note that `git rev-parse --git-path hooks` requires git 2.5+, so we catch the error
+# produced if git is an earlier version and default to `.git/hooks`.
+GITHOOKSDIR := $(shell (git rev-parse --git-path hooks &>/dev/null && git rev-parse --git-path hooks) || echo '.git/hooks')
```

Comment:
Yes, except that it will fail if `.git` is not a directory and Git < 2.5. The hooks directory is not specifically checked for.

So yes, if `.git` is not a directory then we should fail. Anything else would be nonsense. Don't you agree?